plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/perspective/widgets/machine/panel/MachineTreeNode.java

```diff
@@ -48,8 +47,8 @@ public MachineTreeNode(@Assisted MachineTreeNode parent,
 
         boolean isMachine = data instanceof Machine;
 
-        id = isMachine ? ((MachineDto)data).getId() : ROOT;
-        name = isMachine ? ((MachineDto)data).getConfig().getName() : ROOT;
+        id = isMachine ? ((Machine)data).getId() : ROOT;
```

Comment:
@dkuleshov What do you mean? We do check instance type here https://github.com/eclipse/che/pull/2461/files#diff-01a2bd136c2c58a39913908a95a3b6f1R48